net/sched/sch_skbprio.c

```diff
@@ -192,12 +178,11 @@ static int skbprio_change(struct Qdisc *sch, struct nlattr *opt,
 	const unsigned int min_limit = 1;
 
 	if (ctl->limit == (typeof(ctl->limit))-1)
-		q->max_limit = max(qdisc_dev(sch)->tx_queue_len, min_limit);
-	else if (ctl->limit < min_limit ||
-		ctl->limit > qdisc_dev(sch)->tx_queue_len)
+		sch->limit = max(q->max_limit, min_limit);
```

Comment:
Field `max_limit` is not solving a problem. Let's drop it and justing taking away `|| ctl->limit > qdisc_dev(sch)->tx_queue_len)` from the test above.